video/out/vo_gpu_next.c

```diff
@@ -1518,13 +1518,20 @@ static int control(struct vo *vo, uint32_t request, void *data)
         return VO_TRUE;
 
     case VOCTRL_UPDATE_RENDER_OPTS: {
-        m_config_cache_update(p->opts_cache);
         update_ra_ctx_options(vo, &p->ra_ctx->opts);
         if (p->ra_ctx->fns->update_render_opts)
             p->ra_ctx->fns->update_render_opts(p->ra_ctx);
-        update_render_options(vo);
         vo->want_redraw = true;
 
+        // Special case for --image-lut which requires a full
+        // reset.
+        int old_type = p->next_opts->image_lut.type;
+        update_options(vo);
+        struct user_lut image_lut = p->next_opts->image_lut;
+        p->want_reset = image_lut.opt && ((!image_lut.path && image_lut.opt) ||
```

Comment:
Maybe change `=` to `|=` in case something else already sets it, now or in a future change.

Reply:
Good point.